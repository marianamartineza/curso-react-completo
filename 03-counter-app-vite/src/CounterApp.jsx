import { useState } from 'react';
import PropTypes from 'prop-types';

export const CounterApp = ({ value }) => {
  const [counter, setCounter] = useState(value);

  const handleAdd = () => {
    // console.log('+1');
    // value=1000;
    // console.log(value);
    // setCounter(counter + 1);
    setCounter(c => c + 1);
  };

  const handleSubstract = () => setCounter(c => c - 1);

  const handleReset = () => setCounter(value);

  return (
    <>
      <h1>CounterApp</h1>
      <h2 data-testid='test-counter'>{counter}</h2>

      <button aria-label='btn+1' onClick={handleAdd}>
        {' '}
        +1{' '}
      </button>
      <button aria-label='btn-1' onClick={handleSubstract}>
        {' '}
        -1{' '}
      </button>
      <button aria-label='btn-reset' onClick={handleReset}>
        {' '}
        reset{' '}
      </button>
    </>
  );
};

CounterApp.propTypes = {
  value: PropTypes.number.isRequired,
};
