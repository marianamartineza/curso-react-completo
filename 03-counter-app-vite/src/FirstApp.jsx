//obligar 
import PropTypes from 'prop-types';


//import {Fragment} from 'react';

const newMessage = {
    message: 'Hola mundo',
    title: 'Mariana'
};

const getSaludo = (nombre) => {
    return `Hola ${nombre}`;
}

// const saludo = getSaludo('Mariana');
// console.log(saludo);

// if(!title){
//     throw new Error('El title no existe');
// }

export const FirstApp = ({title,subTitle}) => {

    // console.log(props);

    return (
    //<Fragment>
    <>
        <h1>{title}</h1>
        {/* <h1>{getSaludo('Mariana')}</h1> */}
        {/* <h1>Hola Mundo</h1> */}
        {/* <code>{JSON.stringify(newMessage)}</code> */}
        <p>{subTitle + 1}</p>
        <p>Mariana Martinez</p>
    </>
    //</Fragment>
    );
}

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subTitle: PropTypes.number.isRequired
}

// snnipet rafc 

FirstApp.defaultProps = {
    title: 'No hay titulo',
    subTitle: 'No hay subtitulo',
    name: 'Mariana Martinez',
}
