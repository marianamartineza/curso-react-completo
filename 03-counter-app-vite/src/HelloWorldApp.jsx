export function App() {
  // document.creatElement...
  return <h1>Hola Mundo!!!</h1>;
}

export const HelloWorldApp = () => {
  return <h1>Hello World App</h1>;
};
