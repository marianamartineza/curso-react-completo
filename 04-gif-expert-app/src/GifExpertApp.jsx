import {useState} from 'react';
import { AddCategory } from './components/AddCategory';

export const GifExpertApp = () => {
  
  const [categories, setCategories] = useState(['One Punch', 'Dragon Ball']);

  
  const onAddCategory = () => {

    // categories.push('Valorant');
    // console.log('Valorant');
    // setCategories([ ...categories, 'Valorant']);

    setCategories([ 'Valorant', ...categories]);
    
  }

  console.log(categories)
  
    return (
   <>
        {/* titulo */}
        <h1>GiftExpertApp</h1>

        {/* input */}
        <AddCategory setCategories = {setCategories}/>

        {/* listado de Gif */}
        {/* <button onClick={onAddCategory}>Agregar</button> */}



        <ol> 
            {
                categories.map(category => {
                    return <li key={category}>{category}</li>
                })
            }
        </ol>
           
            {/* gif item */}
   </>
  )
}
