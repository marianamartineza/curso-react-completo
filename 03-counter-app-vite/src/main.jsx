import React from 'react';
import ReactDOM from 'react-dom/client';
import './styles.css';

// import {HelloWorldApp} from './HelloWorldApp';

// ReactDOM.createRoot(document.getElementById('root')).render(
//     <React.StrictMode>
//         <HelloWorldApp />
//     </React.StrictMode>
// );

import {FirstApp} from './FirstApp';
// import {CounterApp} from './CounterApp'

ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
        {/* properties - propiedades  */}
        {/* <FirstApp subTitle={1234} /> */}
        {/* <CounterApp value={10} /> */}
         <FirstApp title="Hola, soy vegeta" subTitle={12}/> 
    </React.StrictMode>
)


