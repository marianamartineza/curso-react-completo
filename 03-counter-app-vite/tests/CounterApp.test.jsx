import { fireEvent, render,screen } from "@testing-library/react"
import { CounterApp } from "../src/CounterApp"

describe('pruebas en <CounterApp/>', () => { 

    //tarea 
    const initialValue= 10;

    test('debe de hacer match con el snapshot', () => { 
        const {container}=render(<CounterApp value={initialValue}/>)
        expect(container).toMatchSnapshot();
     });

     test('debe de mostrar el valor inicial de 100', () => { 
        render(<CounterApp value={100}/>)
        expect(screen.getByText(100)).toBeTruthy();
        // expect(screen.getByRole('heading',{level: 2}).innerHTML).toContain("100");
    });

    test('debe de incrementar con el boton +1', () => { 
        render(<CounterApp value={initialValue}/>);
        fireEvent.click(screen.getByText('+1'));
        // screen.debug();
        expect(screen.getByText('11')).toBeTruthy();
     });

     test('debe de decrementar con el boton -1', () => { 
        render(<CounterApp value={initialValue}/>);
        fireEvent.click(screen.getByText('-1'));
        expect(screen.getByText('9')).toBeTruthy();
      });

      test('debe de reiniciar al valor inicial con el boton reset', () => {
        render(<CounterApp value={300}/>);
        fireEvent.click(screen.getByText('+1'));
        fireEvent.click(screen.getByText('+1'));
        fireEvent.click(screen.getByText('+1'));
        // screen.debug();
        // fireEvent.click(screen.getByText('reset'));
        fireEvent.click(screen.getByRole('button',{name: 'btn-reset'}));
        // screen.debug();
        // expect(screen.getByText(300)).toBeTruthy();
        expect(screen.getByTestId('test-counter').innerHTML).toContain('300');
        // expect(getByTestId('test-title').innerHTML).toContain(title);
      });

 });