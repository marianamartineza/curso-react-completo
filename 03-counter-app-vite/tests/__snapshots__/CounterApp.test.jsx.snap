// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <CounterApp/> debe de hacer match con el snapshot 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h2
    data-testid="test-counter"
  >
    10
  </h2>
  <button
    aria-label="btn+1"
  >
     +1 
  </button>
  <button
    aria-label="btn-1"
  >
     -1 
  </button>
  <button
    aria-label="btn-reset"
  >
     reset 
  </button>
</div>
`;
